### PR TITLE
Add Sourcegraph case study.

### PR DESCRIPTION
**Pull request by zchase**
As the title states, this PR adds a Sourcegraph case study.

Issue: https://github.com/pulumi/home/issues/930

### CODE REVIEW

File: content/case-studies/sourcegraph.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+title: Sourcegraph
+description: |
+    Pulumi helps to supercharge Kubernetes deployments at Sourcegraph, ensuring their team has continuous access to the latest builds.
+meta_desc: |
+    Pulumi helps to supercharge Kubernetes deployments at Sourcegraph, ensuring their team has continuous access to the latest builds.
+
+customer_name: Sourcegraph
+customer_logo: /logos/customers/sourcegraph_logo.png
```
Comment (cnunciato) [inline content/case-studies/sourcegraph.md:9]: This is fine, but I'd rather use an SVG for this, and there appears to be one on the Sourcegraph site at https://about.sourcegraph.com/sourcegraph/logo.svg. If you're okay with it, I'll push a PR onto this branch that uses that SVG and replaces the one on the home page also. (It needs a quick adjustment for size.)

File: content/case-studies/sourcegraph.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+customer_url: https://about.sourcegraph.com/
+
+exec_summary: |
+    Sourcegraph selects Pulumi to deploy its latest software on Google Kubernetes Engine - eliminating deployment headaches and ensuring their team has continuous access to the latest builds.
```
[the remark this answers is not in the document]
Comment (zchase) [reply content/case-studies/sourcegraph.md:13]: You've done this a few times and I keep forgetting about the triple dash. I'll do better to remember so we don't keep doing this :)

File: content/case-studies/sourcegraph.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+## Best-in-class Development Tools
+
+Pulumi simplifies Sourcegraph deployments in many ways. One key benefit is that it was far easier to configure and manage Kubernetes using TypeScript and modern development environments instead of writing (and then troubleshooting!) YAML. The team’s new deployment workflow is far simpler than what they previously experienced with text-based configuration. Using Pulumi helps Sourcegraph's team frequently deploy new versions with confidence and eliminates deployment errors that previously meant spending hours debugging.
+
+The main benefits of working with programming languages instead of domain-specific languages include: proper logic constructs and object references to compose interconnected API resources, re-usability of common modules, and clear end-to-end orchestration of infrastructure. Using existing IDEs, compilers, and linters to catch issues and incompatibilities, further boosts team velocity.
+
+By adopting Pulumi, the team is able to achieve their rapid deployment goals while using the same tools that they know and love and they use every day developing Sourcegraph for creating and managing all of their cloud infrastructure needs.
```
[the remark this answers is not in the document]
Comment (zchase) [reply content/case-studies/sourcegraph.md:46]: @isaac-pulumi ☝️

**Comment by cnunciato** [comment]
Added the Sourcegraph SVG and adjusted the spacing on the home-page logos accordingly.

**Pull request merged**